### PR TITLE
checks and recovery functions

### PR DESCRIPTION
**Pull request by adecaro**
This PR adds to ttxdb and auditdb a `Check` function that validates the content of the databases. The checks are customisable via injection. Default checks are provided based on the previous version of CheckTTXDBView.

### CODE REVIEW

File: token/services/network/orion/tokenfetcher.go [hunk 1]
```diff
@@ -7,59 +7,66 @@ SPDX-License-Identifier: Apache-2.0
...
 type tokenFetcher struct {
-	network string
+	network     string
+	viewManager *view2.Manager
 }
 
-func (f *tokenFetcher) QueryTokens(context view.Context, namespace string, IDs []*token.ID) ([][]byte, error) {
-	resBoxed, err := view2.GetManager(context).InitiateView(NewRequestQueryTokensView(f.network, namespace, IDs), context.Context())
+func (f *tokenFetcher) QueryTokens(context context.Context, namespace string, IDs []*token.ID) ([][]byte, error) {
+	resBoxed, err := f.viewManager.InitiateView(NewRequestQueryTokensView(f.network, namespace, IDs), context)
```
[the remark this answers is not in the document]
Comment (alexandrosfilios) [reply token/services/network/orion/tokenfetcher.go:37]: But I see that we call a responder view on the custodian. We don't have a way to call services through sessions, right?

**Review by alexandrosfilios — COMMENTED** [review]
Some comments to consider

File: token/sdk/db/checks.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	return common2.NewChecksService(append(common2.NewDefaultCheckers(a.tmsProvider, a.networkProvider, adb, tdb, id), a.checkers...)), nil
+}
+
+type OwnerCheckServiceProvider struct {
+	tmsProvider     common2.TokenManagementServiceProvider
+	networkProvider common2.NetworkProvider
+	checkers        []common2.NamedChecker
+}
+
+func NewOwnerCheckServiceProvider(in struct {
```
[the remark this answers is not in the document]
Comment (adecaro) [reply token/sdk/db/checks.go:45]: okay

File: token/sdk/dig/sdk.go [hunk 3]
```diff
@@ -123,7 +141,11 @@ func (p *SDK) Install() error {
 		p.Container().Provide(digutils.Identity[*kvs.KVS](), dig.As(new(kvs2.KVS))),
 		p.Container().Provide(identity.NewDBStorageProvider),
 		p.Container().Provide(digutils.Identity[*identity.DBStorageProvider](), dig.As(new(identity2.StorageProvider))),
+		p.Container().Provide(db.NewAuditorCheckServiceProvider),
+		p.Container().Provide(digutils.Identity[*db.AuditorCheckServiceProvider](), dig.As(new(auditor.CheckServiceProvider))),
```
[the remark this answers is not in the document]
Comment (adecaro) [reply token/sdk/dig/sdk.go:145]: to the token-sdk, these are default behaviours that we provide

File: token/services/network/fabric/network.go [hunk 4]
```diff
@@ -337,12 +337,12 @@ func (n *Network) FetchPublicParameters(namespace string) ([]byte, error) {
...
 }
 
-func (n *Network) QueryTokens(context view.Context, namespace string, IDs []*token.ID) ([][]byte, error) {
+func (n *Network) QueryTokens(context context.Context, namespace string, IDs []*token.ID) ([][]byte, error) {
 	return n.tokenQueryExecutor.QueryTokens(context, namespace, IDs)
 }
 
-func (n *Network) AreTokensSpent(c view.Context, namespace string, tokenIDs []*token.ID, meta []string) ([]bool, error) {
-	return n.spentTokenQueryExecutor.QuerySpentTokens(c, namespace, tokenIDs, meta)
+func (n *Network) AreTokensSpent(context context.Context, namespace string, tokenIDs []*token.ID, meta []string) ([]bool, error) {
```
[the remark this answers is not in the document]
Comment (adecaro) [reply token/services/network/fabric/network.go:344]: okay

File: integration/ports.go [hunk 5]
```diff
@@ -60,8 +60,8 @@ var (
 
 	AllTestTypes = []*InfrastructureType{
 		WebSocketNoReplication,
-		LibP2PNoReplication,
-		WebSocketWithReplication,
+		// LibP2PNoReplication,
```
Comment (alexandrosfilios) [inline integration/ports.go:63]: These need to be activated

**Review by alexandrosfilios — CHANGES_REQUESTED** [review]

**Review by alexandrosfilios — APPROVED** [review]
LGTM

**Pull request merged**